### DISCUSSION TITLE
Multiple Monitors

### DISCUSSION THREAD

**Discussion by kentcdodds** [johnlindquist/kit#193] [Ideas]
With Alfred, I can configure the window to pop open on whichever one the mouse is currently. I'd like that with script kit as well :)

**Comment by johnlindquist** [comment]
That's the default behavior.

Any chance you dragged/resized to a different monitor? It caches the last closed position.

If so, try the keyboard shortcut from the monitor the mouse is on, then drag the prompt back so it can cache it properly. I'll look into ways of customizing those behaviors.

**Reply by kentcdodds** [reply]
Cool, so I can drag it and then it'll open where I dragged it. Would be nice if I could tell it to not cache the position and instead just always open on the monitor the mouse is on.

**Reply by johnlindquist** [reply]
It maps "current screen" (determined by the mouse) to a last position/size. So each screen can have a customized position/size based on where the user last left it. 

If that's not what you're experiencing, it's a bug.

I can see how that can causes confusion if the prompt is dragged between screens. I can disable and/or allow user reset that behavior. Or maybe ignore caching if dragged between screens. 🤔

**Reply by kentcdodds** [reply]
Yeah, that's not what's happening for me. As soon as I move it to a position on either screen it remains in that position and on that screen regardless of where the mouse is when I open it in the future.

**Reply by johnlindquist** [reply]
Thanks for letting me know. Investigating...

Can you please post your `~/kenv/cache/prompt.json` below?

**Reply by kentcdodds** [reply]
```json
{
	"prompt": {
		"458642684": {
			"bounds": {
				"x": 3193,
				"y": 25,
				"width": 629,
				"height": 480
			}
		},
		"458650775": {
			"bounds": {
				"x": 3193,
				"y": 25,
				"width": 629,
				"height": 373
			}
		}
	}
}
```

**Reply by johnlindquist** [reply]
Thanks a million. Those are definitely both the same position 😱 (The longers numbers are screen ids). Hmmm...

You can try manually adjusting the cached x,y for the time being 😅

**Reply by johnlindquist** [reply]
If you delete the `~/kenv/cache/prompt.json`, does that fix it? I'm struggling to reproduce and wondering if the cache is from an older version.

**Reply by kentcdodds** [reply]
Nope, no dice :-(

**Reply by johnlindquist** [reply]
Dang. I'll put in some extra screen/cursor/position verification checks over the next couple days to make it a bit more ironclad. Hopefully that will clear it up.

**Reply by kentcdodds** [reply]
Thanks!
